core/trino-main/src/main/java/io/trino/sql/planner/ExpressionInterpreter.java

```diff
@@ -1428,6 +1426,16 @@ private Expression toExpression(Object base, Type type)
         }
     }
 
+    private static boolean isNullLiteralExpression(Object expression)
+    {
+        if (expression == null) {
+            return true;
+        }
+
+        // NULL
+        return expression instanceof NullLiteral;
```

Comment:
In the expression optimization process, i think we should clear design that null value is "optimized" or "evaluated" to Java `null`, so that we don't need to do this.

In fact, i wonder why we check for `NullLiteral`, but we don't check eg for `Cast(NullLiteral, BIGINT)`

Reply:
I've improved this part of the code and now, `Cast(NullLiteral ...)` is checked as well

Reply:
what about the first part of the comment?

> In the expression optimization process, i think we should clear design that null value is "optimized" or "evaluated" to Java null, so that we don't need to do this.

Reply:
actually I've removed the null check in that particular method as it won't occur there (method is called on expression  before optimization)

Reply:
Is it fine? `NullLiteral` is optimized to Java null in `LiteralInterpreter.evaluate`